File: app/src/main/java/org/sil/storyproducer/controller/PagerAdapter.java
package org.sil.storyproducer.controller;

import android.support.v4.app.FragmentManager;
import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentPagerAdapter;

import org.sil.storyproducer.controller.community.CommunityCheckFrag;
import org.sil.storyproducer.controller.consultant.ConsultantCheckFrag;
import org.sil.storyproducer.controller.draft.TransFrag;

public class PagerAdapter extends FragmentPagerAdapter {
    Context context;
    private static int NUM_OF_FRAGS = 5;
    private static int FRAG_TYPE = 0;
    private static String FRAG_STORY = "";
    public PagerAdapter(Context context, FragmentManager fm, int fragNum, int fragType, String fragStory){
        super(fm);
        this.context = context;
        NUM_OF_FRAGS = fragNum;
        FRAG_TYPE = fragType;
        FRAG_STORY = fragStory;
    }

    @Override
    public int getCount() {
        return NUM_OF_FRAGS;
    }

    @Override
    public Fragment getItem(int position) {
        switch (FRAG_TYPE){
            //Translate
            case 1:
                return TransFrag.newInstance(position, NUM_OF_FRAGS, FRAG_STORY);
            //Community
            case 2:
                return new CommunityCheckFrag();
            //Consultant
            case 3:
                return new ConsultantCheckFrag();
        }
        return null;
    }
}

File: app/src/main/java/org/sil/storyproducer/controller/phase/PagerAdapter.kt
package org.sil.storyproducer.controller.phase

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import org.sil.storyproducer.controller.export.CreateFragment
import org.sil.storyproducer.controller.export.ShareFragment
import org.sil.storyproducer.controller.learn.LearnFragment
import org.sil.storyproducer.controller.pager.PagerBaseFragment
import org.sil.storyproducer.controller.remote.WholeStoryBackTranslationFragment
import org.sil.storyproducer.model.PHASE_TYPE
import org.sil.storyproducer.model.PhaseType
import org.sil.storyproducer.model.Workspace

class PagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {

    override fun getItem(i: Int): Fragment {
        val phaseType = Workspace.phases[i]
        val fragment = when (phaseType) {
            PhaseType.LEARN -> LearnFragment()
            PhaseType.CREATE -> CreateFragment()
            PhaseType.WHOLE_STORY -> WholeStoryBackTranslationFragment()
            PhaseType.SHARE -> ShareFragment()
            PhaseType.DRAFT -> PagerBaseFragment()
            PhaseType.COMMUNITY_CHECK -> PagerBaseFragment()
            PhaseType.CONSULTANT_CHECK -> PagerBaseFragment()
            PhaseType.REMOTE_CHECK -> PagerBaseFragment()
            PhaseType.BACKT -> PagerBaseFragment()
            PhaseType.DRAMATIZATION -> PagerBaseFragment()
        }
        val passedArgs = Bundle()
        passedArgs.putInt(PHASE_TYPE, phaseType.ordinal)
        fragment.arguments = passedArgs

        return fragment
    }

    override fun getCount(): Int {
        return Workspace.phases.size
    }

    override fun getPageTitle(position: Int): CharSequence {
        return Workspace.phases[position].getDisplayName()
    }
}
